src/main/kotlin/lotto/core/WinningRank.kt

```diff
@@ -16,7 +16,17 @@ enum class WinningRank(val winningCount: Pair<Int, Boolean>, val winningAmount:
             winningCount: Int,
             matchBonus: Boolean,
         ): WinningRank {
-            return RANK_BY_WINNING_COUNT[winningCount to matchBonus] ?: NOTHING
+            var ranks = entries.filter { it.winningCount == winningCount }
+            if (ranks.isEmpty()) {
+                return NOTHING
+            }
+
+            if (ranks.size == 1) {
+                return ranks[0]
```

Comment:
사소하지만  첫번째 요소를 뽑는거라면 `first()` 확장함수를 활용해볼 수 있을 것 같아요. 😃
마찬가지로 마지막 요소를 뽑는거라면 `last()`를 활용해볼 수 있을 것 같아요. 😉
```suggestion
                return ranks.first()
```

Reply:
네. 생각 못햇네요.
 반영하겠습니다.